ts/input/tex/mathtools/MathtoolsMethods.ts

```diff
@@ -0,0 +1,747 @@
+    const post = parser.GetArgument(name);
+    const body = parser.GetArgument(name);
+    MathtoolsUtil.addPairedDelims(parser.configuration, cs, [open, close, body, n, pre, post]);
+  },
+
+  /**
+   * Implements \centeredcolon, \ordinarycolon, \MTThinColon.
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   * @param {boolean} center     True if colon should be centered
+   * @param {boolean} force      True menas always center (don't use centercolon option).
+   * @param {boolean} thin       True if this is a thin color (for \coloneqq, etc).
+   */
+  CenterColon(parser: TexParser, _name: string, center: boolean, force: boolean = false, thin: boolean = false) {
+    const options = parser.options.mathtools;
+    let mml = parser.create('token', 'mo', {}, ':');
+    if (center && (options['centercolon'] || force)) {
+      const dy = options['centercolon-offset'];
+      mml = parser.create('node', 'mpadded', [mml], {
+        voffset: dy, height: `+${dy}`, depth: `-${dy}`,
+          ...(thin ? {width: options['thincolon-dw'], lspace: options['thincolon-dx']} : {})
+      });
+    }
+    parser.Push(mml);
+  },
+
+  /**
+   * Implements \coloneqq and related macros.
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   * @param {string} tex         The tex string to use (if not using unicode versions or if there isn't one).
+   * @param {string} unicode     The unicode character (if there is one).
+   */
+  Relation(parser: TexParser, _name: string, tex: string, unicode?: string) {
+    const options = parser.options.mathtools;
+    if (options['use-unicode'] && unicode) {
+      parser.Push(parser.create('token', 'mo', {texClass: TEXCLASS.REL}, unicode));
+    } else {
+      tex = '\\mathrel{' + tex.replace(/:/g, '\\MTThinColon').replace(/-/g, '\\mathrel{-}') + '}';
+      parser.string = ParseUtil.addArgs(parser, tex, parser.string.substr(parser.i));
+      parser.i = 0;
+    }
+  },
+
+  /**
+   * Implements \ndownarrow and \nuparrow via a terrible hack (visual only, no chance of this working with SRE).
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   */
+  NArrow(parser: TexParser, _name: string, c: string, dy: string) {
+    parser.Push(
+      parser.create('node', 'TeXAtom', [
+        parser.create('token', 'mtext', {}, c),
+        parser.create('node', 'mpadded', [
+          parser.create('node', 'mpadded', [
+            parser.create('node', 'menclose', [
+              parser.create('node', 'mspace', [], {height: '.2em', depth: 0, width: '.4em'})
+            ], {notation: 'updiagonalstrike', 'data-thickness': '.05em', 'data-padding': 0})
+          ], {width: 0, lspace: '-.5width', voffset: dy}),
+          parser.create('node', 'mphantom', [
+            parser.create('token', 'mtext', {}, c)
+          ])
```

Comment:
So you are overlaying the same arrow twice? That is a bit like poor man's bold (or whatever it is called). 
Is there some future way we could do that "properly" that is also SRE friendly?

Reply:
No, it is not overplayed twice.  Notice that the second is inside an `mphantom`, so it is not visual.  It is only used for positioning purposes (the `space: '-.5width'` below means we are shifting the preceding (mpadded) `menclose` to the left by half the width of the arrow given in `c`.  That is how the "not" line is centered on the arrow (so that it will work in any font, regardless of the width of the arrow).

Reply:
As for making it SRE friendly, there don't seem to be unicode positions for up or down arrows with a diagonal slash (otherwise I'd just have called on those).  The closest is U+2908 and U+2909, which are explicit about having a horizontal slash.